File: backend/app/schemas/comment.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CommentBase(BaseModel):
    body: str


class CommentCreate(CommentBase):
    card_id: uuid.UUID


class CommentRead(CommentBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    card_id: uuid.UUID
    author_id: uuid.UUID
    is_edited: bool
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/comment.ts
export interface Comment {
  id: string;
  body: string;
  card_id: string;
  author_id: string;
  is_edited: boolean;
  created_at: string;
  updated_at: string;
}

export interface CommentCreate {
  body: string;
  card_id: string;
}

export interface CommentUpdate {
  body: string;
}
